receiver/receivercreator/resourceenhancer.go

```diff
@@ -36,23 +36,31 @@ type resourceEnhancer struct {
 
 func newResourceEnhancer(
 	resources resourceAttributes,
+	receiverAttributes map[string]string,
 	env observer.EndpointEnv,
 	endpoint observer.Endpoint,
 	nextConsumer consumer.Metrics,
 ) (*resourceEnhancer, error) {
 	attrs := map[string]string{}
 
-	// Precompute values that will be inserted for each resource object passed through.
-	for attr, expr := range resources[endpoint.Details.Type()] {
-		res, err := evalBackticksInConfigValue(expr, env)
-		if err != nil {
-			return nil, fmt.Errorf("failed processing resource attribute %q for endpoint %v: %w", attr, endpoint.ID, err)
-		}
+	for _, resource := range []map[string]string{resources[endpoint.Details.Type()], receiverAttributes} {
+		// Precompute values that will be inserted for each resource object passed through.
+		for attr, expr := range resource {
+			// If the attribute value is empty this signals to delete existing
+			if expr == "" {
+				delete(attrs, attr)
+				continue
+			}
```

Comment:
We have [similar functionality](https://github.com/open-telemetry/opentelemetry-collector/blob/58e66af2558a56be1f26c759db80e0ed35234607/service/telemetry/config.go#L28-L32) where we use `nil` to delete attributes. Should we probably apply the same approach here?

Reply:
The [existing implementation assumes empty strings aren't desirable for attribute values](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/05f3ece2ce4591aefae4522b3c143f6b3a220e7c/receiver/receivercreator/resourceenhancer.go#L52) and these changes are in line with that precedent. I can see how this may be overly opinionated as is but think it would be a breaking change to adopt. If we want it I'd prefer to shift everything over to `map[string]*string` in a subsequent PR to keep the scope reduced and help with reducing unforeseen side effects.

Reply:
I'm ok with this approach as long as we don't want users to have empty string values